pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -352,6 +354,24 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) er
 	return nil
 }
 
+func retry(pr *v1alpha1.PipelineRun, after *duckv1alpha1.Condition) {
```

Comment:
can this function have a slightly more descriptive name, e.g. `RetryIfNeeded` and have unit tests?

Reply:
maybe you could even make a retry related package and put the logic for both taskruns and pipelineruns into it? would make it easier for the two to share logic if needed

Reply:
This retry (I like that name btw) is very specific to (Tasks|Pipeline)Runs so I guess has more sense to be here as part of reconciliation. I thought that in the inception but this had more sense to be by then. WDYT?